src/main/java/com/exasol/errorreporting/ErrorMessageBuilder.java

```diff
@@ -0,0 +1,158 @@
+    }
+
+    /**
+     * Add a parameter. This method quotes the parameter.
+     * <p>
+     * You can use the parameter in message and mitigation using {@code {{parameter}}}.
+     * </p>
+     * 
+     * @param placeholder placeholder without parentheses
+     * @param value       value to insert
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder parameter(final String placeholder, final Object value) {
+        return this.unquotedParameter(placeholder, Quoter.quoteObject(value));
+    }
+
+    /**
+     * Add a parameter. This method quotes the parameter.
+     * <p>
+     * You can use the parameter in message and mitigation using {@code {{parameter}}}.
+     * </p>
+     * 
+     * @param placeholder placeholder without parentheses
+     * @param value       value to insert
+     * @param description description for the error catalog
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder parameter(final String placeholder, final Object value, final String description) {
+        return this.parameter(placeholder, value);
+    }
+
+    /**
+     * Add a parameter without quotes.
+     * 
+     * @param placeholder placeholder without parentheses
+     * @param value       value to insert
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder unquotedParameter(final String placeholder, final Object value) {
+        this.parameterMapping.put(placeholder, value.toString());
+        return this;
+    }
+
+    /**
+     * Add a parameter without quotes.
+     *
+     * @param placeholder placeholder without parentheses
+     * @param value       value to insert
+     * @param description description for the error catalog
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder unquotedParameter(final String placeholder, final Object value,
+            final String description) {
+        return this.unquotedParameter(placeholder, value);
+    }
+
+    /**
+     * Add a mitigation. Explain here what users can do to resolve or avoid this error.
+     *
+     * @param mitigation explanation
+     * @return self for fluent programming
+     */
+    public ErrorMessageBuilder mitigation(final String mitigation) {
+        this.mitigations.add(mitigation);
+        return this;
+    }
+
+    /**
+     * Build the error message.
+     * 
+     * @return built error massage string
+     */
+    public String toString() {
+        final StringBuilder result = new StringBuilder();
+        result.append(this.errorCode);
+        if (this.messageBuilder.length() > 0) {
+            result.append(": ");
+            result.append(applyPlaceholders(this.messageBuilder.toString()));
+        }
+        if (this.mitigations.size() == 1) {
+            result.append(" ");
+            result.append(this.mitigations.get(0));
+        } else if (this.mitigations.size() > 1) {
+            result.append(" Known mitigations:");
```

Comment:
maybe we should be consistent here and deliver the same structure even if we have 1 or >1 mitigations. What do you think?

Reply:
I prefe it likt this. In most cases we only have only one mitigation. "Like Please create the file". In that cases it degreases the readability.